packages/react-vega/src/VegaEmbed.tsx

```diff
@@ -102,11 +103,11 @@ export default class VegaEmbed extends React.PureComponent<VegaEmbedProps> {
   };
 
   modifyView = (action: ViewListener) => {
-    if (this.viewPromise) {
-      this.viewPromise
-        .then((view) => {
-          if (view) {
-            action(view);
+    if (this.resultPromise) {
+      this.resultPromise
```

Comment:
At some point, we should rewrite this to use a sync/await but that's out of scope for this pull request.

Reply:
i tried to keep the diff to a minimum set of changes